From f0c4046514f41072be20da07b69d8a92220c9294 Mon Sep 17 00:00:00 2001
From: Juhani Numminen <juhaninumminen0@gmail.com>
Date: Thu, 11 Mar 2021 14:56:33 +0200
Subject: [PATCH] include <limits> for numeric_limits

Omission of this include will cause build failures with GCC 11.
https://www.gnu.org/software/gcc/gcc-11/porting_to.html

Original bug report: https://bugs.debian.org/984212
---
 IntervalTree.h         | 1 +
 interval_tree_test.cpp | 1 +
 2 files changed, 2 insertions(+)

diff --git a/IntervalTree.h b/IntervalTree.h
index eb9fc62..ec6640c 100644
--- a/IntervalTree.h
+++ b/IntervalTree.h
@@ -6,6 +6,7 @@
 #include <iostream>
 #include <memory>
 #include <cassert>
+#include <limits>
 
 #ifdef USE_INTERVAL_TREE_NAMESPACE
 namespace interval_tree {
diff --git a/interval_tree_test.cpp b/interval_tree_test.cpp
index fbfee2f..58ec91c 100644
--- a/interval_tree_test.cpp
+++ b/interval_tree_test.cpp
@@ -2,6 +2,7 @@
 #include <thread>
 #include <chrono>
 #include <random>
+#include <limits>
 #include <time.h>
 #include <assert.h>
 #include "IntervalTree.h"
